recipes/mongo_gem.rb

```diff
@@ -0,0 +1,26 @@
+# CopyPaste from edelight/chef-mongodb.
+
+# The build-essential cookbook was not running during the compile phase, install gcc explicitly for rhel so native
+# extensions can be installed
+gcc = package 'gcc' do
+  action :nothing
+  only_if { platform_family?('rhel') }
+end
+gcc.run_action(:install)
+
+if platform_family?('rhel')
+  sasldev_pkg = 'cyrus-sasl-devel'
+else
+  sasldev_pkg = 'libsasl2-dev'
+end
+
```

Comment:
Should make sure that you do a check if ubuntu/debian and if install build-essential. This will avoid issues with building the mongo gem on ubuntu/debian